From b2aefe1463690d98473ffdcfc7d2cd17d9c6f200 Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 10 Aug 2021 10:18:40 +0300
Subject: [PATCH] [SW-51853] gaudi: security: fix dyn cfg block

The PCIe RRs are 32 bits and extend the values with leading 0x7F.
HOST_PHYS_BASE is defined by 0x8000000000 which turns to be 0x7F00000000
in the register. This is not correct max value for the protection we
want.
Fix this by subtracting 1 from the HOST_PHYS_BASE to get a valid max
value.

Change-Id: I56d801fef9c6f6fe17f8e716e2964c590de9fe21
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/arch/gaudi/security.c b/arch/gaudi/security.c
index f377c43..4f4ed3d 100644
--- a/arch/gaudi/security.c
+++ b/arch/gaudi/security.c
@@ -1969,7 +1969,14 @@
 	[CFG_HIGH] = {
 		.rr = PCIE_DYN_RR_CFG_HIGH,
 		.min = mmPSOC_GPIO0_BASE - 1,
-		.max = HOST_PHYS_BASE,
+		/*
+		 * The PCIe RR can take only values of 32 bits, then H/W is
+		 * extending these with 0x7F in the bits 32-39.
+		 * This means we cannot provide the HOST_PHYS_BASE as it will
+		 * just turn into 0x7F00000000 which is not what we want.
+		 * Subtract 1 from HOST_PHYS_BASE to hae a valid value.
+		 */
+		.max = HOST_PHYS_BASE - 1,
 	},
 };
 
